mixer/adapter/rbac/util.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"istio.io/istio/mixer/template/authorization"
+	"istio.io/istio/pkg/log"
+)
+
+// SubjectArgs contains information about the subject of a request.
+type SubjectArgs struct {
+	User       string
+	Groups     string
+	Properties []string
+}
+
+// ActionArgs contains information about the detail of a request.
+type ActionArgs struct {
+	Namespace  string
+	Service    string
+	Method     string
+	Path       string
+	Properties []string
+}
+
+type customLogger struct{}
+
+// Infof from adapter.Logger.
```

Comment:
Why do we need to use a custom logger?

Reply:
Renamed to istioctlLogger and added comment to make it clear it's used in istioctl to redirect the mixer adapter logging to the standard log in istioctl.